contracts/Bonding.sol

```diff
@@ -97,85 +106,138 @@ contract Bonding is CollectableDust {
+
+        // First block 2020 = 9193266 https://etherscan.io/block/9193266
+        // First block 2021 = 11565019 https://etherscan.io/block/11565019
+        // 2020 = 2371753 blocks = 366 days
+        // 1 week = 45361 blocks = 2371753*7/366
+        // n = (block + duration * 45361)
+        // id = n - n % blockRonding
+        // blockRonding = 100 => 2 ending zeros
+        uint256 n = block.number + _weeks * 45361;
+        _id = n - (n % blockRonding);
+
+        _bond(_sharesAmount, _id);
     }
 
-    function redeemShares(uint256 _sharesAmount) public {
-        _updateOracle();
+    function redeemShares(uint256 _sharesAmount, uint256 _id) public {
+        require(
+            block.number > _id,
+            "Bonding: Redeem not allowed before bonding time"
+        );
 
         require(
-            IERC20(manager.bondingShareAddress()).balanceOf(msg.sender) >=
-                _sharesAmount,
+            IERC1155Ubiquity(manager.bondingShareAddress()).balanceOf(
+                msg.sender,
+                _id
+            ) >= _sharesAmount,
             "Bonding: Caller does not have enough shares"
         );
 
-        IBondingShare(manager.bondingShareAddress()).burnFrom(
+        _updateOracle();
+        uint256 _currentShareValue = currentShareValue();
+
+        IBondingShare(manager.bondingShareAddress()).burn(
             msg.sender,
+            _id,
             _sharesAmount
         );
 
-        uint256 tokenAmount =
-            (_sharesAmount * currentShareValue()) / TARGET_PRICE;
+        // uint256 tokenAmount = formulaRedeemBonds(_sharesAmount, _currentShareValue);
 
-        if (redeemStreamTime == 0) {
-            IERC20(manager.uADTokenAddress()).safeTransfer(
-                msg.sender,
-                tokenAmount
-            );
-        } else {
-            // The transaction must be processed by the Ethereum blockchain before
-            // the start time of the stream, or otherwise the sablier contract
-            // reverts with a "start time before block.timestamp" message.
-            uint256 streamStart = block.timestamp + 60; // tx mining + 60 seconds
-
-            uint256 streamStop = streamStart + redeemStreamTime;
-            // The deposit must be a multiple of the difference between the stop
-            // time and the start time
-            uint256 streamDuration = streamStop - streamStart;
-            tokenAmount = (tokenAmount / streamDuration) * streamDuration;
-
-            IERC20(manager.uADTokenAddress()).safeApprove(address(sablier), 0);
-            IERC20(manager.uADTokenAddress()).safeApprove(
-                address(sablier),
-                tokenAmount
-            );
-
-            sablier.createStream(
-                msg.sender,
-                tokenAmount,
-                manager.uADTokenAddress(),
-                streamStart,
-                streamStop
-            );
-        }
+        // console.log("_sharesAmount", _sharesAmount);
```

Comment:
remove all unnecessary comments like line 244-245

Reply:
ok line 181+ and 244+

I left sablier stuff commented for later ?